### PR TITLE
Handle shadow DOM

### PR DESCRIPTION
**Pull request by backflip**
This PR updates the script to find headings in shadow DOM, too.

Additional changes:
- Elements with `role=["presentation"]` are ignored.
- ~~Visibility is detected using `.offsetParent` rather than calculated styles.~~ This has been reverted, see comments.

Tested on https://www.sbb.ch and https://www.swisscom.ch

### CODE REVIEW

File: src/bookmarklet.js [hunk 1]
```diff
@@ -167,33 +167,11 @@ function htmlEntities(str) {
...
-    }
-    el = el.parentElement;
-    try {
-      css = window.getComputedStyle(el);
-    } catch (error) {
-      return true; // happens on window element
-    }
-  }
-  return true;
+  return el.offsetParent !== null;
```
Comment (betabong) [inline src/bookmarklet.js]: I'm fine with all changes except this one. Checking against offsetParent doesn't handle many cases of invisibility to the accessibility tree. Why have you removed the original code (does it have to do with shadow dom compatibility?)

**Review by betabong — CHANGES_REQUESTED** [review]
Hey @backflip thanks for taking the time! Please have a look at my line comment.

File: src/bookmarklet.js (hunk 1 above)
Comment (backflip) [reply src/bookmarklet.js]: @betabong, very good question! I had issues with slotted content and somehow forgot to follow up. This should be fixed now, the only remaining change in there is regarding the previous `el = el.parentElement`.

**Comment by betabong** [comment]
Perfect, thanks for the contribution!

**Pull request merged**